### PR TITLE
Add types to the theme

### PR DESCRIPTION
**Pull request by fariasmateuss**
In this PR: 

- Add types to the theme
- Fix property type gray-dark #5

### CODE REVIEW

File: src/styles/theme.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-export default {
+export const defaultTheme = {
```
Comment (LauraBeatris) [inline src/styles/theme.ts:1]: I think that it's better, in favor of consistency, to name the theme variable according to the filename.

```suggestion
export const theme = {
```

File: src/@types/styled.d.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import 'styled-components';
+import { defaultTheme } from '../styles/theme';
+
+declare module 'styled-components' {
+  export interface DefaultTheme {
```
Comment (LauraBeatris) [inline src/@types/styled.d.ts:5]: Here would be better to extend the whole theme object, in favor of extensibility. In that way, new theme properties won't need to be added manually here.



```suggestion
  export interface DefaultTheme extends typeof defaultTheme {
```

**Review by LauraBeatris — CHANGES_REQUESTED** [review]

File: src/@types/styled.d.ts (hunk 2 above)
Comment (LauraBeatris) [reply src/@types/styled.d.ts:5]: Also, I would like to point out that exporting an empty interface violates the [@typescript-eslint/no-empty-interface](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-empty-interface.md)

Comment (fariasmateuss) [reply src/@types/styled.d.ts]: Got it! Completed changes

File: src/@types/styled.d.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/* eslint-disable @typescript-eslint/no-empty-interface */
+
```
Comment (LauraBeatris) [inline src/@types/styled.d.ts:2]: Is this empty line necessary? 
```suggestion
```

**Review by LauraBeatris — CHANGES_REQUESTED** [review]

**Comment by LauraBeatris** [comment]
@fariasmateuss Thanks for the PR!

**Pull request merged**